consumer/wrapper/storage_wrapper.go

```diff
@@ -0,0 +1,37 @@
+package wrapper
+
+import (
+	"github.com/goat-project/goat-proto-go"
+)
+
+type storageWrapper struct {
+	st goat_grpc.StorageRecord
+}
+
+type storageXML struct {
+}
+
+// NewStorageWrapper wraps given storage in a RecordWrapper
+func NewStorageWrapper(st goat_grpc.StorageRecord) RecordWrapper {
+	return storageWrapper{
+		st: st,
+	}
+}
+
+func (sw storageWrapper) Filename() string {
+	return sw.st.GetRecordID()
+}
+
+func (sw storageWrapper) AsXML() interface{} {
+	// TODO
+	return storageXML{}
+}
+
+func (sw storageWrapper) AsJSON() interface{} {
+	return sw.st
```

Comment:
I think returning an empty structure would be better for methods that are useless for given data type. Or maybe even better some error message like `Not supported by given type` or something like that.

Reply:
Yeah, I agree it makes a lot more sense to do just that. On the other hand, I'd have to change the return type to `interface{}, error`. I'd suggest to return a pointer to the structure and if the conversion is not implemented, return `nil`?

Reply:
Sorry, but I don't understand what you mean by that :smile: